subsys/bluetooth/host/adv.c

```diff
@@ -430,6 +430,20 @@ static inline bool ad_has_name(const struct bt_data *ad, size_t ad_len)
 	return false;
 }
 
+static inline bool ad_is_limited(const struct bt_data *ad, size_t ad_len)
+{
+	size_t i;
+
+	for (i = 0; i < ad_len; i++) {
+		if (ad[i].type == BT_DATA_FLAGS &&
+		    *ad[i].data & BT_LE_AD_LIMITED) {
```

Comment:
I think this is slightly more clear: 

```suggestion
		    ad[i].data[0] & BT_LE_AD_LIMITED) {
```